crates/moon/tests/test_cases/cond_comp.in/moon.test

```diff
@@ -14,12 +14,12 @@
   moonc link-core ./target/wasm-gc/release/build/lib/lib.core ./target/wasm-gc/release/build/main/main.core -main username/hello/main -o ./target/wasm-gc/release/build/main/main.wasm -pkg-config-path ./src/main/moon.pkg.json -pkg-sources username/hello/lib:./src/lib -pkg-sources username/hello/main:./src/main -target wasm-gc
   
   $ moon build --dry-run --target js --sort-input --nostd
-  moonc build-package ./src/lib/all.mbt ./src/lib/js_and_release.mbt ./src/lib/js_or_wasm.mbt ./src/lib/only_js.mbt ./src/lib/wasm_release_or_js_debug.mbt -o ./target/js/release/build/lib/lib.core -pkg username/hello/lib -pkg-sources username/hello/lib:./src/lib -target js -workspace-path .
```

Comment:
I saw this in wonfix list. So this is because legacy implements conditional compilation wrong, is that true?